Kubernetes Installer v2023.01.13-0 release notes (#883)
replicatedhq/replicated-docs: docs/release-notes/rn-kubernetes-installer.md

# Kubernetes Installer Release Notes

## v2023.01.13-0

> 

Reviewer: Add Warning with:

We recommend use v2023.01.13-1 because a regression issue was identified where the .kube/config is not created. If that happens you might can sorted out the problem by copying the kubeconfig to your home directory. 

@kyleprice @laverya @emosbaugh

Reviewer 2: Only thing I recommend adding is an answer to the question: What is the affect on the user if .kube/config is not created? (As in, why should they care if .kube/config is not created? What breaks for them?)

Reviewer: We checked that still possible to access the kURL with kubectl. We exported and use the `/etc/kubernetes/admin.conf` in this way, honest I do NOT think that would impact any user at all. 

That is the reason why all our tests passed with success on this release. See: https://testgrid.kurl.sh/run/PROD-release-v2023.01.13-0-20230113033219. 

**Currently, kURL does not create the kube/config file at all unless the install is done with `cis compliance`**

However, I think that we are trying to be safe as much as possible. Following the new suggestion based on your feedback. 

> We recommend use v2023.01.13-1 instead because a regression issue was identified on this release where the .kube/config might not be created in the home directory. It should **not** affected the ability to access the cluster when you run `bash -l` with `kubectl`. However, if you face problems to connect to the cluster with kubectl or did not find the .kube/config then, you might able to sorted out it by copying the kubeconfig to your home directory (i.e. `$ cp /etc/kubernetes/admin.conf $HOME/.kube/config`) and grating the permission to the `$HOME/.kube/config` file.

Released on January 13, 2023

### New Features {#new-features-v2023-01-13-0}
* Adds [Velero add-on](https://kurl.sh/docs/add-ons/velero) version 1.9.5.

> * feat: kURL execution logs can be found now under /var/log/kurl/.

Reviewer: ```suggestion
* Adds logs for kURL execution which can be found under `/var/log/kurl/`.
```

* Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) version RELEASE.2023-01-02T09-40-09Z.

> * feat: automatic migrates data from Longhorn to OpenEBS.

Reviewer: ```suggestion
* Adds automatic data migration from Longhorn to OpenEBS.
```

* Adds automatic data migration from Longhorn to OpenEBS.
* Adds a migration path for Weave to Flannel on single-node Kubernetes clusters. This migration requires downtime.
* Adds logs for kURL execution which can be found under `/var/log/kurl/`.

### Bug Fixes {#bug-fixes-v2023-01-13-0}
* Fixes an issue where .kube/config would only be created for some install configurations.

> * Improved log info when upgrading from the Rook `1.0.4` to `1.4.9`..

Reviewer: ```suggestion
* Fixes an issue where the process get stuck in failures scenarios by adding timeouts and improving log info when upgrading from the Rook `1.0.4` to `1.4.9`. 
```

> * Adds [Velero add-on](https://kurl.sh/docs/add-ons/velero) version 1.9.5.
> * Adds logs for kURL execution which can be found under `/var/log/kurl/`.
> * Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) version RELEASE.2023-01-02T09-40-09Z.
> * Adds automatic data migration from Longhorn to OpenEBS.
> * Single-node kubernetes clusters can be migrated from the Weave CNI to Flannel. This migration requires downtime.
> * Increase wait timeouts from 10 to 20 minutes when waiting for new versions of Rook and Ceph to roll-out on upgrades.
> * Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) version RELEASE.2023-01-06T18-11-18Z.
> * Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) verison 1.10.8.
> * Adds [Containerd add-on](https://kurl.sh/docs/add-ons/containerd) version 1.6.15.
> 
> ### Bug Fixes {#bug-fixes-v2023-01-13-0}
> * fix: create .kube/config for all installs.
> * add timeout and better log info when upgrades from the rook versions `1.0.4` to `1.4.9` fails.
> * fix: failures into rock multi-upgrades from 1.0.4-14.2.21 to 1.4.9.
> * fix: (for ubuntu) does not allow remove packages when broken packages are sorted out automatically. Therefore, if be required to do removals ask for manual intervention instead.

Reviewer 3: ```suggestion
* Adds [Velero add-on](https://kurl.sh/docs/add-ons/velero) version 1.9.5.
* Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) versions RELEASE.2023-01-02T09-40-09Z and RELEASE.2023-01-06T18-11-18Z.
* Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) verison 1.10.8.
* Adds [Containerd add-on](https://kurl.sh/docs/add-ons/containerd) version 1.6.15.
* Adds automatic data migration from Longhorn to OpenEBS.
* Adds a migration path for Weave to Flannel on single-node Kubernetes clusters. This migration requires downtime.
* Adds logs for kURL execution which can be found under `/var/log/kurl/`.

### Bug Fixes {#bug-fixes-v2023-01-13-0}
* Fixes an issue where .kube/config would only be created for some install configurations.
* Improved log info when upgrading from the Rook `1.0.4` to `1.4.9`..
* Fixes upgrading Rook from `1.0.4-14.2.21` to `1.4.9`.
* Fixes a bug on Ubuntu where the installer would sometimes remove packages when attempting to install Kubernetes.
* Fixes a timeout waiting for new versions of Rook and Ceph to roll out on upgrades by increase wait timeouts from 10 to 20 minutes.

```
